sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -547,7 +545,38 @@ export class EventHubReceiver extends LinkEntity {
     return result;
   }
 
-  protected _deleteFromCache(): void {
+  /**
+   * Registers the user's onMessage and onError handlers.
+   * Sends buffered events from the queue before adding additional credits to the AMQP link.
+   * @param onMessage
+   * @param onError
+   * @param abortSignal
+   * @ignore
+   */
+  registerHandlers(onMessage: OnMessage, onError?: OnError, abortSignal?: AbortSignalLike): void {
+    if (typeof onMessage !== "function") {
+      // throw error
+    }
+    if (onError && typeof onError !== "function") {
+      // throw error
+    }
```

Comment:
The above checks are not needed. The required checks are already being made in `EventhubConsumer` class